Android/app/src/main/java/io/github/project_travel_mate/login/LoginActivity.java

```diff
@@ -188,17 +192,26 @@ public void onClick(View view) {
                 String confirmPassString = confirm_pass_signup.getText().toString();
                 String firstname = firstName.getText().toString();
                 String lastname = lastName.getText().toString();
-                if (validateEmail(emailString)) {
-                    if (validatePassword(passString)) {
-                        if (passString.equals(confirmPassString)) {
-                            mLoginPresenter.ok_signUp(firstname, lastname, emailString, passString, mHandler);
-                        } else {
-                            Snackbar snackbar = Snackbar
-                                    .make(findViewById(android.R.id.content),
-                                          R.string.passwords_check, Snackbar.LENGTH_LONG);
-                            snackbar.show();
+                if (!firstname.isEmpty() && !lastname.isEmpty()) {
```

Comment:
instead of adding these many checks here, add a function similar to validateEmail to validate firstname & lastname

Reply:
Done